tensorboard/plugins/pr_curve/pr_curves_plugin.py

```diff
@@ -80,24 +83,75 @@ def pr_curves_impl(self, runs, tag):
     Returns:
       The JSON object for the PR curves route response.
     """
-    response_mapping = {}
-    for run in runs:
-      try:
-        tensor_events = self._multiplexer.Tensors(run, tag)
-      except KeyError:
-        raise ValueError(
-            'No PR curves could be fetched for run %r and tag %r' % (run, tag))
-
-      content = self._multiplexer.SummaryMetadata(run, tag).plugin_data.content
-      pr_curve_data = metadata.parse_plugin_metadata(content)
-      thresholds = [
-          float(v) / pr_curve_data.num_thresholds
-          for v in range(1, pr_curve_data.num_thresholds + 1)]
-
-      response_mapping[run] = [
-          self._process_tensor_event(e, thresholds) for e in tensor_events]
+    if self._db_connection_provider:
+      # Serve data from the database.
+      db = self._db_connection_provider()
+
+      # We select for steps greater than -1 because the writer inserts
+      # placeholder rows en masse. The check for step filters out those rows.
```

Comment:
Please do this using `dtype IS NOT NULL` instead, in accordance with the [schema definition](https://github.com/tensorflow/tensorflow/blob/4d9bef194799c0656c9ea80f8f083cfab1ebc14a/tensorflow/contrib/tensorboard/db/schema.cc#L92).

Reply:
Thanks! Much more straightforward. Good to know the guarantees in the schema!

Reply:
#1081